anta/models.py

```diff
@@ -505,13 +505,22 @@ def save_commands_data(self, eos_data: list[dict[str, Any] | str]) -> None:
             self.instance_commands[index].output = data
 
     def __init_subclass__(cls) -> None:
-        """Verify that the mandatory class attributes are defined."""
-        mandatory_attributes = ["name", "description", "categories", "commands"]
+        """Verify that the mandatory class attributes are defined and set name and description if not set."""
+        mandatory_attributes = ["categories", "commands"]
         for attr in mandatory_attributes:
             if not hasattr(cls, attr):
                 msg = f"Class {cls.__module__}.{cls.__name__} is missing required class attribute {attr}"
                 raise NotImplementedError(msg)
```

Comment:
Old code but I think `AttributeError` is more appropriate here + collect all missing attributes and report them together provides a clearer error message and helps users fix all issues in one go.

```suggestion
    missing_attrs = [attr for attr in mandatory_attributes if not hasattr(cls, attr)]
    if missing_attrs:
        attrs_str = ', '.join(missing_attrs)
        msg = f"Class {cls.__module__}.{cls.__name__} is missing required class attribute(s): {attrs_str}"
        raise AttributeError(msg)
```